gdb/gdb.go

```diff
@@ -1,198 +1,285 @@
+}
+
+func getCachedGenres(ctx context.Context, apikey string) map[string]gamesdb.Genre {
+	genres := genresCache.genres
+	if genres != nil {
+		return *genres
+	}
+
+	genresCache.mux.Lock()
+	defer genresCache.mux.Unlock()
+
+	genres = genresCache.genres
+	if genres == nil {
+		apiGenres := getGenres(ctx, apikey)
+		genres = &apiGenres
+		genresCache.genres = genres
+	}
+
+	return *genres
 }
 
-// OImage is used to parse <Images> tags for fanart and
-// screenshots since they are formatted differently.
-type OImage struct {
-	Original Image  `xml:"original"`
-	Thumb    string `xml:"thumb"`
+// ParsedDeveloper is a normalized GamesDB Developer
+type ParsedDeveloper struct {
+	ID   int
+	Name string
 }
 
-// Game is used to parse the GetGame's <Game> tag.
-type Game struct {
-	ID          string `xml:"id"`
-	GameTitle   string
-	Overview    string
+// ParsedGenre is a normalized GamesDB Genre
+type ParsedGenre struct {
+	ID   int
+	Name string
+}
+
+// ParsedPublisher is a normalized GamesDB Publisher
+type ParsedPublisher struct {
+	ID   int
+	Name string
+}
+
+// ParsedGameImage is a normalized GamesDB GameImage
+type ParsedGameImage struct {
+	ID       int
+	Type     string
+	Side     string
+	Filename string
+}
+
+// ParsedImageSizeBaseUrls is a normalized GamesDB ImageBaseUrlMeta
+type ParsedImageSizeBaseUrls struct {
+	Original string
+	Thumb    string
+}
+
+// ParsedGame is  a normalized GamesDB Game
+type ParsedGame struct {
+	ID          int
+	Name        string
 	ReleaseDate string
-	Platform    string
-	Developer   string
-	Publisher   string
-	Genres      []string `xml:"Genres>genre"`
-	Players     string
-	Rating      float64
-	ESRB        string
-	AltTitles   []string `xml:"AlternateTitles>title"`
-	BoxArt      []Image  `xml:"Images>boxart"`
-	ClearLogo   []Image  `xml:"Images>clearlogo"`
-	Banner      []Image  `xml:"Images>banner"`
-	FanArt      []OImage `xml:"Images>fanart"`
-	Screenshot  []OImage `xml:"Images>screenshot"`
+	//Platform    int
+	Players    int
+	Overview   string
+	Developers []ParsedDeveloper
+	Genres     []ParsedGenre
+	Publishers []ParsedPublisher
+
+	Images        map[string][]ParsedGameImage
+	ImageBaseUrls ParsedImageSizeBaseUrls
 }
 
 // GetGame gets the game information from the DB.
-func GetGame(ctx context.Context, req GGReq) (*GGResp, error) {
-	u, err := url.Parse(gdbURL)
-	u.Path = ggPath
-	q := url.Values{}
-	switch {
-	case req.ID != "":
-		q.Set("id", req.ID)
-	case req.Name != "":
-		q.Set("name", req.Name)
-		if req.Platform != "" {
-			q.Set("platform", req.Platform)
-		}
-	default:
+func GetGame(ctx context.Context, apikey string, gameID string) (*ParsedGame, error) {
+	var games gamesdb.GamesByGameId
+	var resp *http.Response
+	var err error
+
+	// TODO(jpr): remove unneeded fields
+	//fields := "players,publishers,genres,overview,last_updated,rating,platform,coop,youtube,os,processor,ram,hdd,video,sound,alternates"
+	fields := "players,publishers,genres,overview,platform"
+
+	if gameID != "" {
```

Comment:
More idiomatic Go would be be to reduce the indent of the good code path. Also since we don't have the name or ID you can simplify the further.
```
if gameID == "" {
  return nil, fmt.Errorf("must provide an ID or Name")
}
games, resp, err := apiClient.GamesApi.GamesByGameID(ctx, apikey, gameID, &gamesdb.GamesByGameIDOpts{Fields: optional.NewString(fields)})
if err != nil
  return nil, fmt.Errorf("getting game url:%s, error:%s", resp.Request.URL, err)
}
```

Reply:
Fixed